src/bindings/jna/libelektra/src/main/java/org/libelektra/KeySet.java

```diff
@@ -422,6 +500,85 @@ public Iterator<Key> iterator() {
     return new KeySetIterator<>(this, Key::new);
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @return New {@link DescendingKeySetView} backed by this {@link KeySet}
+   */
+  @Override
+  public NavigableSet<Key> descendingSet() {
+    return new DescendingKeySetView(this, null, true, null, false);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return New {@link DescendingKeySetIterator} backed by this {@link KeySet}
+   */
+  @Override
+  public Iterator<Key> descendingIterator() {
+    return this.descendingSet().iterator();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @throws IllegalStateException if this {@link KeySet} has already been released
+   * @throws NullPointerException if {@code fromElement} or {@code toElement} is {@code null}
+   * @throws IllegalArgumentException {@inheritDoc}
+   */
+  @Override
+  public NavigableSet<Key> subSet(
+      Key fromElement, boolean fromInclusive, Key toElement, boolean toInclusive) {
+    requireNonNull(fromElement);
+    requireNonNull(toElement);
+
+    var fromElementIndex = indexOf(fromElement);
+    var toElementIndex = indexOf(toElement);
```

Comment:
@kodebach I think this is line with the NavigableSet#subSet java doc stating:
```
@throws IllegalArgumentException if {@code fromElement} is
     *         greater than {@code toElement}; or if this set itself
     *         has a restricted range, and {@code fromElement} or
     *         {@code toElement} lies outside the bounds of the range.
```

Reply:
I think the way to interpret this is

```java
SortedSet<Int> myset; // a set containing integers 1,2,4,5

myset.subSet(2,5); // returns set with 2,4
myset.subSet(4,10); // might throw, but IMO should return set with 4,5

SortedSet<Int> subset = myset.subSet(2,5);
subset.subSet(4, 10); // should throw, because subset is restricted to 2 <= x < 5 and 10 > 5

// I also think to be consistent that this should also throw IllegalArgumentException
subset.add(10); // throw because 10 is out of bounds

// In any case, both of these should work
myset.subSet(2, 4); // returns set with just 2, because there are no bounds
subset.subSet(2, 4); // returns set with just 2, because subset is restricted to 2 <= x < 5 and 2 <= 2 < 5 and 2 <= 4 < 5

// finally these should also work, because otherwise how do you get the end bit of a set?
myset.subSet(2, 6); // returns set with 4, 5
subset.subSet(2, 5); // returns set with just 4
```

In other words, you cannot use `SortedSet#subSet` or `NavigableSet#subSet` to move the bounds of a set "outwards". You can only leave them unchanged or move them "inwards". A set that was not constructed by `subSet` has infinite bounds.